server/src/main/java/org/apache/druid/server/coordinator/loading/SegmentLoadingConfig.java

```diff
@@ -136,8 +121,35 @@ public int getMaxReplicaAssignmentsInRun()
     return maxReplicaAssignmentsInRun;
   }
 
-  public int getPercentDecommSegmentsToMove()
+  public int getBalancerComputeThreads()
   {
-    return percentDecommSegmentsToMove;
+    return balancerComputeThreads;
+  }
+
+  /**
+   * Computes the number of threads to be used in the balancing executor.
+   * The number of used segments in a cluster is generally a good indicator of
+   * the cluster size and has been used here as a proxy for the actual number of
+   * segments that would be involved in cost computations.
+   * <p>
+   * The number of threads increases by 1 first for every 50k segments, then for
+   * every 75k segments and so on.
+   *
+   * @return Number of {@code balancerComputeThreads} in the range [1, 8].
+   */
+  public static int computeNumBalancerThreads(int numUsedSegments)
```

Comment:
Nit: would it better to consider the number of replicas rather than the number of used segments in cases where a lot of segments have 0 replicas in the load rule

Reply:
The javadoc calls it out that we are using `numUsedSegments` as a proxy for the number of replicas. If we count number of replicas at this point, we might under-estimate the number of segments as some unavailable segments might not have been assigned yet (since this is before the `RunRules` duty). This might lead to too few threads.